core/src/main/java/org/apache/iceberg/TableMetadata.java

```diff
@@ -1521,6 +1521,11 @@ public Builder setPreviousFileLocation(String previousFileLocation) {
       return this;
     }
 
+    public Builder setLastUpdatedMillis(long lastUpdatedMillis) {
+      this.lastUpdatedMillis = lastUpdatedMillis;
+      return this;
+    }
```

Comment:
Is this just for DDL or DML like add snapshot too ? 
if its for adding snapshots too,
 then do we need to make sure this is called before addSnapshot() / setRef and handle the setRef that don't take timestamp fom the snapshot obj ? 
https://github.com/apache/iceberg/blob/f65558ec30a27fb26ed85ba0e3dc4e498f2c046b/core/src/main/java/org/apache/iceberg/TableMetadata.java#L1317

if we call this after addSnapshot() / setRef then my snapshot logs would reference different time than the metadata log ? 
https://github.com/apache/iceberg/blob/f65558ec30a27fb26ed85ba0e3dc4e498f2c046b/core/src/main/java/org/apache/iceberg/TableMetadata.java#L1326

if its just for DDL do we need to assert its just for ops which doesn't add snapshot ?

Reply:
That's a good callout, currently the SnapshotLogEntry and MetadataLogEntry share the same lastUpdatedMillis and want to minimize the change here. Updated to only set if existing value is null